### ISSUE TITLE
Apparent discrepancy in the definition of a block between rules

### ISSUE THREAD

**Issue by kboucher** [eslint/eslint#16131]
### Environment

Node version: 16.14.1
npm version: 8.5.0
Local ESLint version: 7.32.0
Global ESLint version: N/A
Operating System: OS X

### What parser are you using?

@babel/eslint-parser

### What did you do?

For the `lines-around-comment` rule, setting `allowBlockStart: true` is not respected for `switch` statements. 

According to this issue: https://github.com/eslint/eslint/issues/9653, that is because `switch` statement curly braces are not considered "blocks". See comment: https://github.com/eslint/eslint/issues/9653#issuecomment-347386313

However, the `padded-blocks` rule appears to interpret `switch` statement curly braces as blocks which results in a discrepancy that can only be remedied by never including a comment inside a `switch` statement's opening curly brace.

Relevant rule configurations:
```js
'lines-around-comment': [
    2,
    {
        allowArrayStart: true,
        allowBlockStart: true,
        allowObjectStart: true,
        beforeBlockComment: true,
        beforeLineComment: true,
    },
],
'padded-blocks': ['error', 'never'],
```

Sample code
```js
let someValue = "a";

switch (someValue) {
    // some comment
  case 'a':
    someValue = "b";
    break;
  
  default:
    someValue = "c";
}
```

Link to reproduction of issue:
https://eslint.org/play/#eyJ0ZXh0IjoibGV0IHNvbWVWYWx1ZSA9IFwiYVwiO1xuXG5zd2l0Y2ggKHNvbWVWYWx1ZSkge1xuICAgIC8vIHNvbWUgY29tbWVudFxuICBjYXNlICdhJzpcbiAgICBzb21lVmFsdWUgPSBcImJcIjtcbiAgICBicmVhaztcbiAgXG4gIGRlZmF1bHQ6XG4gICAgc29tZVZhbHVlID0gXCJjXCI7XG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6eyJsaW5lcy1hcm91bmQtY29tbWVudCI6WzIseyJhbGxvd0Jsb2NrU3RhcnQiOnRydWUsImJlZm9yZUxpbmVDb21tZW50Ijp0cnVlLCJiZWZvcmVCbG9ja0NvbW1lbnQiOnRydWUsImFmdGVyQmxvY2tDb21tZW50IjpmYWxzZSwiYWZ0ZXJMaW5lQ29tbWVudCI6ZmFsc2UsImFsbG93QmxvY2tFbmQiOmZhbHNlfV0sInBhZGRlZC1ibG9ja3MiOlsiZXJyb3IiLCJuZXZlciJdfSwiZW52Ijp7ImVzNiI6dHJ1ZX19fQ==

### What did you expect to happen?

I expected that setting `allowBlockStart: true` would preclude a violation of `beforeLineComment: true` in the `lines-around-comment` rule.

Given the discussion in issue 9653, I would then expect `'padded-blocks': ['error', 'never']` to not result in an error inside a `switch` statement's opening curly brace.

### What actually happened?

To visualize, visit the reproduction and add a line break before the comment. Note that the `lines-around-comment` error becomes a `padded-blocks` error because the two rules do not consistently interpret `switch` case curly braces as blocks.

https://eslint.org/play/#eyJ0ZXh0IjoibGV0IHNvbWVWYWx1ZSA9IFwiYVwiO1xuXG5zd2l0Y2ggKHNvbWVWYWx1ZSkge1xuICAgIFxuICAgIC8vIHNvbWUgY29tbWVudFxuICBjYXNlICdhJzpcbiAgICBzb21lVmFsdWUgPSBcImJcIjtcbiAgICBicmVhaztcbiAgXG4gIGRlZmF1bHQ6XG4gICAgc29tZVZhbHVlID0gXCJjXCI7XG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6eyJsaW5lcy1hcm91bmQtY29tbWVudCI6WzIseyJhbGxvd0Jsb2NrU3RhcnQiOnRydWUsImJlZm9yZUxpbmVDb21tZW50Ijp0cnVlLCJiZWZvcmVCbG9ja0NvbW1lbnQiOnRydWUsImFmdGVyQmxvY2tDb21tZW50IjpmYWxzZSwiYWZ0ZXJMaW5lQ29tbWVudCI6ZmFsc2UsImFsbG93QmxvY2tFbmQiOmZhbHNlfV0sInBhZGRlZC1ibG9ja3MiOlsiZXJyb3IiLCJuZXZlciJdfSwiZW52Ijp7ImVzNiI6dHJ1ZX19fQ==

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by nzakas** [comment]
Yeah, that definitely seems inconsistent. Unfortunately, these things happen when you have hundreds of rules to manage.

It seems like the best way forward here would be to add the options suggested in https://github.com/eslint/eslint/issues/9653#issuecomment-347386313: allowSwitchStart and allowSwitchEnd. 

Let’s see what the rest of the team thinks.

**Comment by kboucher** [comment]
> Unfortunately, these things happen when you have hundreds of rules to manage.

Totally not judging. 😁 

I would just like to be able to add a comment before the first `case` in a `switch` statement without having to turn off the `padded-blocks` rule.

Looking forward to a possible solution. Thanks!

**Comment by mdjermanovic** [comment]
`allowBlockEnd: true` already applies to `switch` statements so it would be consistent that `allowBlockStart: true` applies as well.

```js
/* eslint lines-around-comment: ["error", {
  beforeLineComment: true,
  afterLineComment: true,
  allowBlockStart: true,
  allowBlockEnd: true
}]*/

switch (foo) {
  // this comment isn't allowed by allowBlockStart: true 
    
  case 1:    
    bar();
    break;
    
  // this comment is allowed by allowBlockEnd: true
}

```

[Playground link](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IGxpbmVzLWFyb3VuZC1jb21tZW50OiBbXCJlcnJvclwiLCB7XG4gIGJlZm9yZUxpbmVDb21tZW50OiB0cnVlLFxuICBhZnRlckxpbmVDb21tZW50OiB0cnVlLFxuICBhbGxvd0Jsb2NrU3RhcnQ6IHRydWUsXG4gIGFsbG93QmxvY2tFbmQ6IHRydWVcbn1dKi9cblxuc3dpdGNoIChmb28pIHtcbiAgLy8gdGhpcyBjb21tZW50IGlzbid0IGFsbG93ZWQgYnkgYWxsb3dCbG9ja1N0YXJ0OiB0cnVlIFxuICAgIFxuICBjYXNlIDE6ICAgIFxuICAgIGJhcigpO1xuICAgIGJyZWFrO1xuICAgIFxuICAvLyB0aGlzIGNvbW1lbnQgaXMgYWxsb3dlZCBieSBhbGxvd0Jsb2NrRW5kOiB0cnVlXG59XG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7ImVzNiI6dHJ1ZX19fQ==)

**Comment by nzakas** [comment]
Oh wow, didn’t realize that. In that case, I agree it’s a bug and we should fix it so allowBlockStart also applies to switch statements.

**Comment by snitin315** [comment]
I'll  prepare a fix for it.

**Comment by nzakas** [comment]
Thanks @snitin315

**Comment by kboucher** [comment]
Thank you @snitin315!

**Comment by snitin315** [comment]
@kboucher the next release is scheduled on 29th July.
